### DISCUSSION TITLE
Include JXL as Foreign TIFF Compression

### DISCUSSION THREAD

**Discussion by AngelaDMerkel** [libvips/libvips#3818] [Ideas]
Libvips struggles to properly handle metadata when moving between TIFF and JXL. Allowing JXL as foreign compression much like it does JPEG would help to realise many of the lossless space savings of JXL alongside the wide feature set of TIFF— without needing to compromise on metadata quality. 
There is some precedent for this by Adobe. Adobe is using JXL compression inside their DNG files when performing a photomerge in Ligthtroom based on the logic of space savingings in linear DNG files.

**Comment by jcupitt** [comment]
Yes, I'd support this. I think tiled tiff with JXL compression would be a great format, and I wish the JXL devs would add something like that. 

There's already a JXL compression tag in libtiff, you probably saw:

https://gitlab.com/libtiff/libtiff/-/blob/master/libtiff/tiff.h?ref_type=heads#L218

**Reply by AngelaDMerkel** [reply]
I've seen that tag, but as far as I know there isn't an easy way to handle the creation of such files. Pyramidal tiling is incredibly useful in TIFF and as far as I can tell [GeoTIFFs](https://developers.planet.com/docs/planetschool/an-introduction-to-cloud-optimized-geotiffs-cogs-part-1-overview/) [do support JXL compression](https://gdal.org/drivers/raster/cog.html#cog-cloud-optimized-geotiff-generator) (?). There is just no easy way to produce a more mundane tiff with JXL that I am aware of. 
Although Adobe doesn't officially support JXL compression in their TIFF standard I see no reason why their products and others wouldn't decode just fine.

**Reply by AngelaDMerkel** [reply]
@jcupitt Is there still consideration for adding this, or are you waiting for further upstream development of `cjxl`?

**Reply by jcupitt** [reply]
Hi again @AngelaDMerkel, sure, let's make this into an enhancement issue.

**Reply by jcupitt** [reply]
Enhancement issue https://github.com/libvips/libvips/issues/3951
